File: typescript/examples/agents/experimental/remote.ts
import "dotenv/config.js";
import { Client as MCPClient } from "@i-am-bee/acp-sdk/client/index.js";
import { SSEClientTransport } from "@i-am-bee/acp-sdk/client/sse.js";
import { RemoteAgent } from "beeai-framework/agents/experimental/remote/agent";
import { createConsoleReader } from "examples/helpers/io.js";
import { FrameworkError } from "beeai-framework/errors";

const agentName = "literature-review";

const instance = new RemoteAgent({
  client: new MCPClient({
    name: "example-remote-agent",
    version: "1.0.0",
  }),
  transport: new SSEClientTransport(new URL("/mcp/sse", "http://127.0.0.1:8333")),
  agent: agentName,
});

const reader = createConsoleReader();

try {
  for await (const { prompt } of reader) {
    const result = await instance.run({ prompt: JSON.parse(prompt) }).observe((emitter) => {
      emitter.on("update", (data) => {
        reader.write(`Agent (received progress) 🤖 : `, data.output);
      });
    });

    reader.write(`Agent (${agentName}) 🤖 : `, result.message.text);
  }
} catch (error) {
  reader.write("Agent (error)  🤖", FrameworkError.ensure(error).dump());
}

File: python/examples/agents/experimental/remote.py
import asyncio
import json
import sys
import traceback

from beeai_framework.agents.experimental.remote.agent import RemoteAgent
from beeai_framework.errors import FrameworkError
from examples.helpers.io import ConsoleReader


async def main() -> None:
    reader = ConsoleReader()

    agent = RemoteAgent(url="http://127.0.0.1:8333/mcp/sse", agent="chat")
    for prompt in reader:
        # Run the agent and observe events
        response = await agent.run(
            prompt=f'{{"messages":[{{"role":"user","content":"{prompt}"}}],"config":{{"tools":["weather","search"]}}}}'
        ).on(
            "update",
            lambda data, event: (
                reader.write("Agent 🤖 : ", data["update"]["value"]["logs"][0]["message"])
                if "logs" in data["update"]["value"]
                else None
            ),
        )

        reader.write("Agent 🤖 : ", json.loads(response.result.text)["messages"][0]["content"])


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except FrameworkError as e:
        traceback.print_exc()
        sys.exit(e.explain())
